File: frontend/src/types/job.ts
export type JobMode = "general" | "jd_targeted";
export type JobStatus = "queued" | "parsing" | "structuring" | "optimizing" | "diffing" | "completed" | "failed";

export type CreateJobResponse = {
  job_id: string;
  status: JobStatus;
  current_step: string;
  progress: number;
  created_at: string;
};

export type JobStatusResponse = {
  job_id: string;
  status: JobStatus;
  current_step: string;
  progress: number;
  error_message: string | null;
  warnings: string[];
  created_at: string;
  updated_at: string;
};

export type CreateJobInput = {
  resume_file: File;
  mode: JobMode;
  job_description?: string;
  target_role?: string;
  notes?: string;
};

File: backend/app/schemas/job.py
from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


JobMode = Literal["general", "jd_targeted"]
JobStatus = Literal[
    "queued",
    "parsing",
    "structuring",
    "optimizing",
    "diffing",
    "completed",
    "failed",
]


class CreateJobResponse(BaseModel):
    job_id: str
    status: JobStatus
    current_step: str
    progress: int
    created_at: datetime


class JobStatusResponse(BaseModel):
    job_id: str
    status: JobStatus
    current_step: str
    progress: int
    error_message: str | None = None
    warnings: list[str] = Field(default_factory=list)
    created_at: datetime
    updated_at: datetime
